src/Raven.Server/Documents/Queries/LuceneIntegration/TermsMatchQuery.cs

```diff
@@ -69,7 +68,7 @@ public override bool Next(IState state)
                         {
                             break; // search the next term
                         }
-                    }
+                    } while (termEnum.Next(state));
```

Comment:
This isn't correct, you remain on the same term that you are already know is too small